GetStream/stream-chat-android
# [CAS-1177] Use partial updates when pinning message

https://stream-io.atlassian.net/browse/CAS-1177

### 🎯 Goal

Use partial message update when pinning/unpinning a message

### 🧪 Testing

1. Ensure that a channel member has the permission to `PinMessage`

2. Pin/unpin a message by using the code below:

```kotlin
// Contains just the message ID
val message = Message(id = message.id)

// Pin the message for 10 minutes
ChatClient.instance().pinMessage(message = message, Date(System.currentTimeMillis() + 600_000)).enqueue()
// Pin the message for 10 minutes
ChatClient.instance().pinMessage(message = message, 600).enqueue()
// Unpin the message
ChatClient.instance().unpinMessage(message = message).enqueue()
```

3. Wait for the `MessageUpdatedEvent` event and ensure that `Message.pinned`, `Message.pinnedAt`, `Message.pinExpires`, `Message.pinnedBy` fields contain the correct values. Also, the data in other `Message` fields should not be lost (for example, the `Message.text` field should contain the original message text).

### ☑️ Checklist
